stan/math/prim/prob/neg_binomial_2_lpmf.hpp

```diff
@@ -79,33 +84,60 @@ return_type_t<T_location, T_precision> neg_binomial_2_lpmf(
   }
 
   for (size_t i = 0; i < max_size_seq_view; i++) {
-    if (include_summand<propto>::value) {
-      logp -= lgamma(n_vec[i] + 1.0);
-    }
-    if (include_summand<propto, T_precision>::value) {
-      logp += multiply_log(phi__[i], phi__[i]) - lgamma(phi__[i]);
-    }
-    if (include_summand<propto, T_location>::value) {
-      logp += multiply_log(n_vec[i], mu__[i]);
-    }
-    if (include_summand<propto, T_precision>::value) {
-      logp += lgamma(n_plus_phi[i]);
-    }
-    logp -= (n_plus_phi[i]) * log_mu_plus_phi[i];
-
-    // if phi is large we probably overflow, defer to Poisson:
-    if (phi__[i] > 1e5) {
-      logp = poisson_lpmf(n_vec[i], mu__[i]);
-    }
-
-    if (!is_constant_all<T_location>::value) {
-      ops_partials.edge1_.partials_[i]
-          += n_vec[i] / mu__[i] - (n_vec[i] + phi__[i]) / (mu__[i] + phi__[i]);
-    }
-    if (!is_constant_all<T_precision>::value) {
-      ops_partials.edge2_.partials_[i]
-          += 1.0 - n_plus_phi[i] / (mu__[i] + phi__[i]) + log_phi[i]
-             - log_mu_plus_phi[i] - digamma(phi__[i]) + digamma(n_plus_phi[i]);
+    if (phi__[i] > internal::neg_binomial_2_phi_cutoff) {
+      // Phi is large, deferring to Poisson.
+      // Copying the code here as just calling
+      // poisson_lpmf does not preserve propto logic correctly.
+      // Note that Poisson can be seen as first term of Taylor series for
+      // phi -> Inf. Similarly, the derivativew wrt. mu and phi can be obtained
```

Comment:
Typo: "derivativew". Also I would use "wrt" rather than "wrt." as the latter makes me think the sentence is finished (there are other uses).